584 soft client-side authentication

kaitmore opened the pull request:
I ended up just checking whether a service was authorized in FabricRouter and conditionally rendering the InstanceRouter or GMServiceView based on that.

I used a UI kit notification to indicate a redirect just for now until we factor all of that out.

Resolves #584

spmcbride1201 commented on src/components/fabric/FabricGrid.js:
```diff
@@ -35,7 +36,13 @@ class FabricGrid extends Component {
     // Refresh services from the Fabric Server every time this loads
+    const { location: { state } } = this.props;
     Actions.fetchServices();
+    if (state && !state.authorized)
```
Please add some documentation explaining that this state is added by the router.

spmcbride1201 commented on src/components/fabric/FabricGrid.js:
```diff
@@ -35,7 +36,13 @@ class FabricGrid extends Component {
+    if (state && !state.authorized)
+      notification("Authorization required", {
+        status: "danger",
+        timeout: 500
```
This feels way too quick to me. I'd consider keeping it on the screen and forcing the user to dismiss it.

spmcbride1201 replied:
The message would be a bit nicer if it was like "You are not authorized to view (insert microservice name here)". That would take some changes to your state.

kaitmore commented:
@spmcbride1201 I edited the notification to include the service name and changed the cursor to 'not-allowed'. Also added comments to clarify where state is coming from in the `FabricGrid`

kaitmore commented:
@spmcbride1201 links are now disabled for unauthorized services

spmcbride1201 commented on src/components/Main/scenes/Fabric/components/FabricMainView/scene/Table/components/GMServiceList/components/GMServiceListItem.js:
```diff
@@ -48,6 +48,12 @@ const ItemVersion = styled.div`
   align-self: center;
 `;
 
+const ItemInfo = GMLink.withComponent("div").extend`
```
Please refactor GMServiceListItem.js into a directory names GMServiceListItem with the logic in index.js and all of these components in a components subdirectory

kaitmore commented:
@spmcbride1201 Refactored `GMServiceListItem` and `GMServiceCard` into new structure
